### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by olensmar**

### CODE REVIEW

File: CONCEPTS.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
+# Swagger Assert4j Core Concepts
+
+## JSON Recipes
+
+JSON recipes are the internal representation used by swagger-assert4j to describe API tests. A very simple example:
```
Comment (spindelmanne) [inline CONCEPTS.md]: Technically it's an "external representation". Should we just mention that you can also write them yourself as an Assert4J user?

File: CONCEPTS.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  ]
+}
+````
+
+A recipe consists of an array of testSteps that can execute requests, transfer values, define properties, etc - each containing 
+properties corresponding to their type. For a quick reference of all available steps and their properties have a look at [REFERENCE.md].
+
+## Creating recipes with code
+
+Creating elaborate recipes in JSON with many teststeps, assertions, transfers, etc.. can be tedious - which is
```
Comment (spindelmanne) [inline CONCEPTS.md]: "etc.." -> "etc"

File: CONCEPTS.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+```java
+RecipeExecutor executor = new SoapUIRecipeExecutor();
+Execution execution = executor.executeRecipe(recipe);
+```
+
+Read more about local execution in the [local module](modules/local).
+
+## Remote Execution
+
+To get access to extended functionality like data-driven testing, centralized execution and reporting, etc., you
```
Comment (spindelmanne) [inline CONCEPTS.md]: "etc." -> "etc"

File: CONCEPTS.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    GET("https://api.swaggerhub.com/apis")
+        .withAssertions(
+            statusCodes(200)
+        ) 
+);
+```
+
+## Execution modes
+
+No matter how you create JSON recipes; manually, via code, auto-generate, etc - they can be executed either locally
```
Comment (spindelmanne) [inline CONCEPTS.md]: Replace the semicolon with a dash

File: README.md [hunk 5]
```diff
@@ -1,11 +1,16 @@
 # Swagger Assert4J - a Java library for API testing
 
-The Assert4J library lets you test APIs through Java, Groovy or Cucumber. The library has extensive support for REST, SOAP, JDBC and JMS protocols. Under the hood the library uses the test-execution engine of [SoapUI](http://www.soapui.org).
-
-Read on to get started
-* [with Java](#getting-started-with-java) - together with any testing framework
-* [with Groovy](#getting-started-with-groovy) - together with any testing framework 
-* [with Cucumber](modules/cucumber) - with cucumber for java 
+The Assert4J library lets you test APIs through Java, Groovy or Cucumber. The library has extensive support for REST, SOAP, JDBC and JMS protocols.
```
Comment (spindelmanne) [inline README.md]: Unfortunately I don't think the library has support JMS test steps yet. It probably should have though.

File: MODULES.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# Swagger Assert4j Modules
+
+Overview of included modules and how they fit together:
+
+| Module | Description |
+|---|---|
+| [core](modules/core)  | Core Recipe creation and execution classes and interfaces |
+| [cucumber](modules/cucumber)  | Extensible Cucumber vocabulary for REST API testing, including commandline and Docker execution |
```
Comment (spindelmanne) [inline MODULES.md]: I think "commandline" should be two words

File: CONCEPTS.md [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+you can use submitRecipe(TestRecipe) instead; the returned Execution object will return 
+ProjectResultReport.StatusEnum.RUNNING until the test finishes (either passed or failed). 
+
+## Result handling
+
+In all instances above we got an Execution object when executing the recipe - getting the actual result
+of the execution is done via execution.getExecutionResult() - which returns an ExecutionResult object that
+provides details on execution time, individual teststep results, etc.
+
+In a unit-testing scenario the provided AssertionUtils class can be used to assert the outcome of an Excution:
```
Comment (spindelmanne) [inline CONCEPTS.md]: "unit-testing" -> "unit testing"; Excution -> Execution

File: README.md [hunk 8]
```diff
@@ -14,70 +19,43 @@ Read on to get started
...
-Usage of the facade as in the above examples also enables logging of both generated recipes and HTTP transaction logs 
-of executed tests (in HAR file format). Adding the following two properties:
-
-```
-assert4j.log.executions.folder=target/logs/executions
-assert4j.log.recipes.folder=target/logs/recipes
-```
-
-will automatically result in the corresponding artifacts being written to the corresponding folders.
+Learn more about the java testing vocabulary by:
```
Comment (spindelmanne) [inline README.md]: Capitalize Java

File: README.md (hunk 5 above)
Comment (spindelmanne) [inline README.md]: I think "using" would be better than "through"

File: README.md [hunk 9]
```diff
@@ -14,70 +19,43 @@ Read on to get started
...
-                     .withAssertions(
-                         json("$.totalCount", "4")
-                     )                 
-                 );
-    
-            assertExecutionResult(result);
-        }
+    @Test
+    public void simpleCountTest() throws Exception {
+         RecipeExecutionResult result = executeRecipe("Simple Count Test",
```
Comment (spindelmanne) [inline README.md]: I understand that you want to label the recipe in this way, to comment on what it's doing. However, this makes the API look more complex than it is, so I'd prefer if we didn't pass the parameter.

File: README.md [hunk 10]
```diff
@@ -14,70 +19,43 @@ Read on to get started
...
+                     json("$.totalCount", "4")
+                 )
+         );
+         
+         assertExecutionResult(result);
+    }
 	```
 
-3. Run your test.
+3. Run your test and enjoy the results (or not...);
```
Comment (spindelmanne) [inline README.md]: Gotta love that change :-)

File: modules/cucumber/README.md [hunk 11]
```diff
@@ -44,76 +43,73 @@ dependency to your pom:
...
+included glue-code, all other options are passed as usual, see https://cucumber.io/docs/reference/jvm#cli-runner.
 
 (you will need java8 installed on your path)
 
 ### Running with Docker
 
-If the above two options are too much of a java-hassle for you, then you can use the corresponding docker image 
-available at https://hub.docker.com/r/smartbear/cucumber4apis instead - it packages the above runner and makes it
-super-easy to run feature files for your APIs, for example:
+If the above two options are too much of a java-hassle for you, then you can use the prepackaged docker image
```
Comment (spindelmanne) [inline modules/cucumber/README.md]: java-hassle -> Java hassle

Also, the "then" isn't necessary

File: modules/cucumber/README.md [hunk 12]
```diff
@@ -44,76 +43,73 @@ dependency to your pom:
...
-specify that that volume as the source for feature files for the Cucumber Runner (together with the -p pretty argument).
+Here we mounted a local folder containing feature files into a volume named "/features" in the container - and then 
+specify that volume as the source for feature files for the Cucumber Runner, together with the -p pretty argument.
 
 ### Recipe logging
 
-If you add a `-Dtestserver.cucumber.logfolder=...` system property to your command line invocation the runner will write 
-generated json recipe files to the specified folder before sending them to the TestServer, for example allowing you 
-to import them into Ready API for load-testing/monitoring/etc.
+If you add a `-Dswagger-assert4j.cucumber.logfolder=...` system property to your command line invocation the runner will write
```
Comment (spindelmanne) [inline modules/cucumber/README.md]: "add a" -> "add the"

File: modules/cucumber/README.md [hunk 13]
```diff
@@ -44,76 +43,73 @@ dependency to your pom:
...
+Here we mounted a local folder containing feature files into a volume named "/features" in the container - and then 
+specify that volume as the source for feature files for the Cucumber Runner, together with the -p pretty argument.
 
 ### Recipe logging
 
-If you add a `-Dtestserver.cucumber.logfolder=...` system property to your command line invocation the runner will write 
-generated json recipe files to the specified folder before sending them to the TestServer, for example allowing you 
-to import them into Ready API for load-testing/monitoring/etc.
+If you add a `-Dswagger-assert4j.cucumber.logfolder=...` system property to your command line invocation the runner will write 
+generated json recipes to the specified folder before executing them, for example allowing you
```
Comment (spindelmanne) [inline modules/cucumber/README.md]: json -> JSON

File: modules/cucumber/README.md [hunk 14]
```diff
@@ -44,76 +43,73 @@ dependency to your pom:
...
+specify that volume as the source for feature files for the Cucumber Runner, together with the -p pretty argument.
 
 ### Recipe logging
 
-If you add a `-Dtestserver.cucumber.logfolder=...` system property to your command line invocation the runner will write 
-generated json recipe files to the specified folder before sending them to the TestServer, for example allowing you 
-to import them into Ready API for load-testing/monitoring/etc.
+If you add a `-Dswagger-assert4j.cucumber.logfolder=...` system property to your command line invocation the runner will write 
+generated json recipes to the specified folder before executing them, for example allowing you 
+to import them into ReadyAPI for load-testing/monitoring/etc.
```
Comment (spindelmanne) [inline modules/cucumber/README.md]: load-testing/monitoring/etc. -> load-testing, monitoring, etc.

File: modules/cucumber/README.md [hunk 15]
```diff
@@ -44,76 +43,73 @@ dependency to your pom:
...
 
-### Configuring Ready! API TestServer access
+### Configuring excution with ReadyAPI TestServer
```
Comment (spindelmanne) [inline modules/cucumber/README.md]: excution -> execution

File: modules/cucumber/README.md [hunk 16]
```diff
@@ -322,17 +316,17 @@ public class CustomStepDefs {
 To get this used during execution you will need to
 
 1. Compile the above into a jar file
-2. Include the jar file in the classpath for the TestServer Cucumber runner
+2. Include the jar file in the classpath for the Cucumber runner
```
Comment (spindelmanne) [inline modules/cucumber/README.md]: jar -> JAR, maybe?

File: modules/cucumber/modules/core/README.md [hunk 17]
```diff
@@ -1,4 +1,4 @@
-## Utility classes for creating Cucumber StepDefs for Ready! API TestServer
+## Utility classes for creating Cucumber StepDefs for swagger-assert4j-cucumber
 
-Have a look at the declarative Cucumber sample in the TestServer samples project how these classes
-can be used to create your own Gherkin vocabulary running against the TestServer
\ No newline at end of file
+Have a look at the declarative Cucumber sample in the [samples module](../samples) how these classes
```
Comment (spindelmanne) [inline modules/cucumber/modules/core/README.md]: "how these" -> "to see how these"

**Comment by olensmar** [comment]
thanks @spindelmanne, great comments - hope I didn't miss anything while fixing!

File: README.md [hunk 18]
```diff
@@ -1,6 +1,6 @@
 # Swagger Assert4J - a Java library for API testing
 
-The Assert4J library lets you test APIs through Java, Groovy or Cucumber. The library has extensive support for REST, SOAP, JDBC and JMS protocols. 
+The Assert4J library lets you test APIs using Java, Groovy or Cucumber. The library has extensive support for REST, SOAP and JDBC protocols.
```
Comment (spindelmanne) [inline README.md]: One more comment @olensmar - technically only SOAP is a protocol. REST is more like a convention I suppose, and JDBC is an API (in the classic sense).

Comment (olensmar) [reply README.md]: sure - how about just having "The Assert4J library lets you easily test REST and SOAP services using Java, Groovy or Cucumber." as the first sentence - the JDBC support is there for helping REST/SOAP testing - and not really for JDBC testing in its own right!?

Comment (spindelmanne) [reply README.md]: Sounds good, as long as we highlight the JDBC support elsewhere.

**Pull request merged**